lib/infrastructure/pigeons/scan_provider.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:cobble/domain/entities/pebble_scan_device.dart';
+import 'package:cobble/infrastructure/pigeons/pigeons.dart';
+import 'package:hooks_riverpod/all.dart';
+
+class ScanState {
+  final bool scanning;
+  final List<PebbleScanDevice> devices;
+
+  ScanState(this.scanning, this.devices);
+}
+
+class _ScanCallbacks extends StateNotifier<ScanState> implements ScanCallbacks {
+  _ScanCallbacks()
+      : super(
+          ScanState(
+            true,
+            [],
+          ),
+        );
+
+  @override
+  void onScanStarted() {
+    state = ScanState(true, state.devices);
+  }
+
+  @override
+  void onScanStopped() {
+    state = ScanState(false, state.devices);
+  }
+
+  @override
+  void onScanUpdate(ListWrapper arg) {
+    final devices = (arg.value.cast<Map>())
+        .map((element) => PebbleScanDevice.fromMap(element))
+        .toList();
+    state = ScanState(state.scanning, devices);
+  }
+}
+
+final scanProvider = StateNotifierProvider<_ScanCallbacks>((ref) {
```

Comment:
Is it a good idea to expose provider of private class?

Reply:
Why not? Users of providers don't need to create new _ScanCallbacks and they won't need to reference its type. All they need from this file is scanProvider and ScanState, no reason to expose more than that.

Reply:
Users can still use _ScanCallbacks like this
`final lastValue = useProvider(pairProvider).stream.last;`

Reply:
If users wish to mock scanProvider, they will need to implement _ScanCallbacks and they will need access to it. I'm going to expose it 👍